File: codegen/results/v002/m3-max-48gb/lmstudio/qwen_qwen3-coder-30b_nothink/20260828_113901/typescript/007_beatles_interview/solution.ts
import fs from 'fs';
import path from 'path';

// Helper function to calculate age as of July 1, 2025
function calculateAge(birthday: string): number {
  const birthDate = new Date(birthday);
  const targetDate = new Date('2025-07-01');
  
  let age = targetDate.getFullYear() - birthDate.getFullYear();
  const monthDiff = targetDate.getMonth() - birthDate.getMonth();
  
  if (monthDiff < 0 || (monthDiff === 0 && targetDate.getDate() < birthDate.getDate())) {
    age--;
  }
  
  return age;
}

// Helper function to parse date from MM/DD/YYYY format
function parseDate(dateStr: string): string {
  const [month, day, year] = dateStr.split('/');
  return `${year}-${month.padStart(2, '0')}-${day.padStart(2, '0')}`;
}

// Read and parse CSV file
const csvData = fs.readFileSync(path.join('input', 'input.csv'), 'utf-8');

// Split into lines and parse headers
const lines = csvData.split('\n').filter(line => line.trim() !== '');
const headers = lines[0].split(',').map(h => h.trim());

// Process each data line
const result = lines.slice(1).map(line => {
  const values = line.split(',').map(v => v.trim());
  
  // Create person object
  const person: any = {
    FirstName: values[0].split(' ').filter(n => n.length > 0)[0],
    LastName: values[0].split(' ').filter(n => n.length > 0).slice(1).join(' '),
    Birthday: parseDate(values[1]),
    Age: calculateAge(parseDate(values[1])),
    Relatives: []
  };
  
  // Add relatives
  const relatives = ['Father', 'Mother', 'Brother', 'Sister'];
  for (let i = 3; i < Math.min(values.length, 7); i++) {
    if (values[i] !== 'null' && values[i].trim() !== '') {
      const relativeName = values[i].trim();
      const relationship = relatives[i - 3];
      
      // Split name into first and last name
      const parts = relativeName.split(' ');
      const relative: any = {
        FirstName: parts[0],
        LastName: parts.slice(1).join(' '),
        Relationship: relationship
      };
      
      person.Relatives.push(relative);
    }
  }
  
  return person;
});

// Output as JSON
console.log(JSON.stringify(result, null, 2));

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen_qwen3-coder-30b_nothink/20260828_113901/python/007_beatles_interview/solution.py
import csv
from datetime import datetime
import json

# Calculate age as of July 1, 2025
reference_date = datetime(2025, 7, 1)

# Read the CSV file
with open('input/input.csv', 'r') as file:
    reader = csv.DictReader(file)
    
    result = []
    
    for row in reader:
        # Parse birthday
        birthday = datetime.strptime(row['Birthday'], '%m/%d/%Y')
        
        # Calculate age
        age = reference_date.year - birthday.year
        if reference_date.month < birthday.month or (reference_date.month == birthday.month and reference_date.day < birthday.day):
            age -= 1
        
        # Build relatives list
        relatives = []
        
        # Add father
        if row['Father'] != 'null':
            father_name = row['Father'].split()
            relatives.append({
                "FirstName": father_name[0],
                "LastName": father_name[1] if len(father_name) > 1 else "",
                "Relationship": "Father"
            })
        
        # Add mother
        if row['Mother'] != 'null':
            mother_name = row['Mother'].split()
            relatives.append({
                "FirstName": mother_name[0],
                "LastName": mother_name[1] if len(mother_name) > 1 else "",
                "Relationship": "Mother"
            })
        
        # Add brother
        if row['Brother'] != 'null':
            brother_name = row['Brother'].split()
            relatives.append({
                "FirstName": brother_name[0],
                "LastName": brother_name[1] if len(brother_name) > 1 else "",
                "Relationship": "Brother"
            })
        
        # Add sister
        if row['Sister'] != 'null':
            sister_name = row['Sister'].split()
            relatives.append({
                "FirstName": sister_name[0],
                "LastName": sister_name[1] if len(sister_name) > 1 else "",
                "Relationship": "Sister"
            })
        
        # Split name into first and last name
        full_name = row['Name'].split()
        first_name = full_name[0]
        last_name = full_name[-1]
        
        # Create person object
        person = {
            "FirstName": first_name,
            "LastName": last_name,
            "Birthday": birthday.strftime("%Y-%m-%d"),
            "Age": age,
            "Relatives": relatives
        }
        
        result.append(person)

# Output JSON array to stdout
print(json.dumps(result, indent=2))